src/Dashboard/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import books from './books';
+import './style.css';
+import logo from '../assets/search.svg';
+import BooksList from './components/BookList';
+
+class Dashboard extends Component {
+  state = {
+    value: '',
+    filter: '',
+    submittedValue: '',
+    submittedFilter: ''
+  }  
+  
+  handleSelectChange = (event) => {
+    this.setState({ filter: event.target.value});
+  }
+
+  handleInputChange = (event) => {
+    this.setState({ value: event.target.value});
+  }
+
+  handleClick = (event) => {
+    if (event.key === 'Enter' | event.type === 'click') {
+      this.setState({submittedValue: this.state.value, submittedFilter: this.state.filter });    
```

Comment:
`setState` is asynchronous, so when you need to set state based on a prev state you can pass setState a function that receives the previousState and the props (you don't need them here) and returns the new state.

Same goes for reading state after setting it. If you need to use the "new" state after a setState call you can pass a callback. (it doesn't apply here)

Read more here https://reactjs.org/docs/react-component.html#setstate